var/spack/repos/builtin/packages/openblas/package.py

```diff
@@ -39,9 +39,14 @@ def install(self, spec, prefix):
             make_defs += ['BUILD_LAPACK_DEPRECATED=1']
 
         # Add support for OpenMP
-        # Note: Make sure your compiler supports OpenMP
         if '+openmp' in spec:
-            make_defs += ['USE_OPENMP=1']
+            # Note: Apple's Clang 7.3.0 does not support OpenMP.
+            # For now, raise an error if we compile with clang on OSX.
+            # TODO: one could still have a self-compiled clang-omp on OSX...
+            if (sys.platform == 'darwin') and (self.compiler.name == 'clang'):
```

Comment:
i agree that a check whether the compiler supports OpenMP is the right thing to do, but it is on the different scale.

For now, it would be fine to make it work on native Clang. I will update the PR so that it hopefully does not break the usage of OpenMP with a non-native Clang.